bundles/net.tourbook/src/net/tourbook/ui/views/TourNutritionView.java

```diff
@@ -0,0 +1,1514 @@
+import net.tourbook.data.TourNutritionProduct;
+import net.tourbook.database.TourDatabase;
+import net.tourbook.nutrition.DialogCustomTourNutritionProduct;
+import net.tourbook.nutrition.NutritionUtils;
+import net.tourbook.nutrition.QuantityType;
+import net.tourbook.preferences.ITourbookPreferences;
+import net.tourbook.tour.ITourEventListener;
+import net.tourbook.tour.SelectionDeletedTours;
+import net.tourbook.tour.SelectionTourData;
+import net.tourbook.tour.SelectionTourId;
+import net.tourbook.tour.TourEvent;
+import net.tourbook.tour.TourEventId;
+import net.tourbook.tour.TourManager;
+import net.tourbook.ui.views.nutrition.DialogSearchProduct;
+import net.tourbook.web.WEB;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.e4.ui.di.PersistState;
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.layout.PixelConverter;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.viewers.CellEditor;
+import org.eclipse.jface.viewers.CellLabelProvider;
+import org.eclipse.jface.viewers.CheckboxCellEditor;
+import org.eclipse.jface.viewers.ColumnViewer;
+import org.eclipse.jface.viewers.ComboBoxCellEditor;
+import org.eclipse.jface.viewers.EditingSupport;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredContentProvider;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerCell;
+import org.eclipse.jface.viewers.ViewerComparator;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Spinner;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Widget;
+import org.eclipse.ui.IPartListener2;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.forms.widgets.ExpandableComposite;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+import org.eclipse.ui.forms.widgets.Section;
+import org.eclipse.ui.part.PageBook;
+import org.eclipse.ui.part.ViewPart;
+
+import cop.swt.widgets.viewers.table.celleditors.RangeContent;
+import cop.swt.widgets.viewers.table.celleditors.SpinnerCellEditor;
+
+public class TourNutritionView extends ViewPart implements ITourViewer {
```

Comment:
The name column is not sorted alphabetically

![image](https://github.com/mytourbook/mytourbook/assets/1283445/daa84799-97b9-4c05-bf75-a5fb496bf29f)

Reply:
It works for me, can you try again ?
![image](https://github.com/mytourbook/mytourbook/assets/25235665/4fdabda6-8804-4250-a5b5-534e380de0ab)

Reply:
> I just saw for the first time in your screenshot the checkboxes in the beverage column, in the dark theme they are almost not visible

You should display a text or an icon, look in these views

![image](https://github.com/mytourbook/mytourbook/assets/1283445/08a3a50d-6410-4f24-a624-3ae94ababc47)

Reply:
>I just saw for the first time in your screenshot the checkboxes in the beverage column, in the dark theme they are almost not visible

Fixed. I chose the icon that looks better

![image](https://github.com/mytourbook/mytourbook/assets/25235665/4c8a800a-53d1-4541-938d-17d74d88eaab)